src/engine/AbsoluteValueConstraint.cpp

```diff
@@ -71,17 +73,25 @@ PiecewiseLinearFunctionType AbsoluteValueConstraint::getType() const
     return PiecewiseLinearFunctionType::ABSOLUTE_VALUE;
 }
 
-PiecewiseLinearConstraint *AbsoluteValueConstraint::duplicateConstraint() const
+ContextDependentPiecewiseLinearConstraint *AbsoluteValueConstraint::duplicateConstraint() const
 {
     AbsoluteValueConstraint *clone = new AbsoluteValueConstraint( _b, _f );
     *clone = *this;
+    this->initializeDuplicatesCDOs( clone );
     return clone;
 }
 
 void AbsoluteValueConstraint::restoreState( const PiecewiseLinearConstraint *state )
 {
     const AbsoluteValueConstraint *abs = dynamic_cast<const AbsoluteValueConstraint *>( state );
+
+    CVC4::context::CDO<bool> *activeStatus = _cdConstraintActive;
```

Comment:
So this is not fantastic logic, it is there simply to avoid losing context objects and causing memory leaks. Once we switch to context-dependent architecture the restoreState methods all over the place will be obsolete.

Reply:
I can re-initialize their values to whatever we are restoring to, but it is really just paranoid code and not likely to be relevant in practice. Alternatively, I can replace current logic with assertions to indicate that the constraint is not in context-dependent mode.